programs/uxd/src/instructions/redeem.rs

```diff
@@ -160,16 +163,66 @@ struct DepositoryInfoForRedeem<'info> {
 
 pub(crate) fn handler(ctx: Context<Redeem>, redeemable_amount: u64) -> Result<()> {
     // Gather all the onchain states we need (caps, weights and supplies)
-    let controller = ctx.accounts.controller.load()?;
+    let mut controller = ctx.accounts.controller.load_mut()?;
     let identity_depository = ctx.accounts.identity_depository.load()?;
     let mercurial_vault_depository = ctx.accounts.mercurial_vault_depository.load()?;
     let credix_lp_depository = ctx.accounts.credix_lp_depository.load()?;
+    let now_timestamp = Clock::get()?.unix_timestamp;
+
+    // Limit per day (NO MERGE, this include both options!! we only need one!!)
```

Comment:
I think for the Zeno paradox thing mentioned earlier... We need both? And we can keep both parameterised, although need proper commenting to explain why

Reply:
For sure, added support for both BPS and amount (and doing the max of the two)

This should solve this issue and let us be quite configurable here